### PR TITLE
feat: create user and auth modules

### PR DESCRIPTION
**Pull request by raphael-foliveira**
Cria os módulos de autorização/autenticação e User

### CODE REVIEW

**Comment by raphael-foliveira** [comment]
> SonarCloud Quality Gate failed.    [Quality Gate failed](https://sonarcloud.io/dashboard?id=nicklima-amigos_football-mm&pullRequest=57)
> 
> [Bug](https://sonarcloud.io/project/issues?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=SECURITY_HOTSPOT) [E](https://sonarcloud.io/project/security_hotspots?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=SECURITY_HOTSPOT) [2 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=CODE_SMELL) [0 Code Smells](https://sonarcloud.io/project/issues?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=CODE_SMELL)
> 
> [No Coverage information](https://sonarcloud.io/component_measures?id=nicklima-amigos_football-mm&pullRequest=57) No Coverage information [](https://sonarcloud.io/component_measures?id=nicklima-amigos_football-mm&pullRequest=57&metric=new_duplicated_lines_density&view=list) [4.9% Duplication](https://sonarcloud.io/component_measures?id=nicklima-amigos_football-mm&pullRequest=57&metric=new_duplicated_lines_density&view=list)
> 
> idea Catch issues before they fail your Quality Gate with our IDE extension sonarlint [SonarLint](https://www.sonarsource.com/products/sonarlint/features/connected-mode/?referrer=sonarcloud-welcome)

@nicolas-limadev Consegue me ajudar com isso?
Ele está detectando credenciais colocadas diretamente no código, só que são apenas credenciais fake pra usar nos testes.

**Comment by nicolas-limadev** [comment]
> > SonarCloud Quality Gate failed.    [Quality Gate failed](https://sonarcloud.io/dashboard?id=nicklima-amigos_football-mm&pullRequest=57)
> > [Bug](https://sonarcloud.io/project/issues?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=SECURITY_HOTSPOT) [E](https://sonarcloud.io/project/security_hotspots?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=SECURITY_HOTSPOT) [2 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=CODE_SMELL) [0 Code Smells](https://sonarcloud.io/project/issues?id=nicklima-amigos_football-mm&pullRequest=57&resolved=false&types=CODE_SMELL)
> > [No Coverage information](https://sonarcloud.io/component_measures?id=nicklima-amigos_football-mm&pullRequest=57) No Coverage information [](https://sonarcloud.io/component_measures?id=nicklima-amigos_football-mm&pullRequest=57&metric=new_duplicated_lines_density&view=list) [4.9% Duplication](https://sonarcloud.io/component_measures?id=nicklima-amigos_football-mm&pullRequest=57&metric=new_duplicated_lines_density&view=list)
> > idea Catch issues before they fail your Quality Gate with our IDE extension sonarlint [SonarLint](https://www.sonarsource.com/products/sonarlint/features/connected-mode/?referrer=sonarcloud-welcome)
> 
> @nicolas-limadev Consegue me ajudar com isso? Ele está detectando credenciais colocadas diretamente no código, só que são apenas credenciais fake pra usar nos testes.

Resolvido.

File: package.json [hunk 1]
```diff
@@ -81,4 +84,4 @@
     "coverageDirectory": "../coverage",
     "testEnvironment": "node"
   }
-}
+}
```
Comment (nicolas-limadev) [inline package.json:87]: Linha abaixo

File: src/app.module.ts [hunk 2]
```diff
@@ -22,6 +24,7 @@ import { LeagueModule } from './league/league.module';
           password: configService.get('DATABASE_PASSWORD'),
```
Comment (nicolas-limadev) [inline src/app.module.ts:24]: Onde fica armazenado essas variáveis de credenciais do database?

Comment (raphael-foliveira) [reply src/app.module.ts:24]: Por enquanto, estão vindo do `docker-compose.yml`. Essas variáveis vão vir do ambiente mesmo.

File: package.json [hunk 3]
```diff
@@ -84,4 +84,4 @@
     "coverageDirectory": "../coverage",
     "testEnvironment": "node"
   }
-}
\ No newline at end of file
+}
```
Comment (raphael-foliveira) [inline package.json:87]: Obrigado @nicolas-limadev .
Não conferi esse arquivo porque geralmente não editamos manualmente, mas realmente estava faltando.

**Review by nicolas-limadev — APPROVED** [review]

**Pull request merged**